From 4425d3e373425e953236b8a8c281812e102e53d1 Mon Sep 17 00:00:00 2001
Message-Id: <4425d3e373425e953236b8a8c281812e102e53d1.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Wed, 30 Jul 2014 15:06:49 +0530
Subject: [PATCH 239/312] FOR_UPSTREAM [VPG]: drm/i915: Re-initialize init
 panel power sequencer during enable DP.

Reusing existing power sequnceri for port and pipe is causing panel power sequnce to
timeout causing the link training to fail. Let's properly re-initialize the pipe & power
sequencer for port currently in use.

Change-Id: I4d627385622d6199a75a2d709db78041bb19e353
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c |    3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 1066a45..3355fcd 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -2330,9 +2330,6 @@ static void vlv_init_panel_power_sequencer(struct intel_dp *intel_dp)
 	struct intel_crtc *crtc = to_intel_crtc(encoder->base.crtc);
 	struct edp_power_seq power_seq;
 
-	if (intel_dp->pipe == crtc->pipe)
-		return;
-
 	/*
 	 * If another power sequencer was being used on this
 	 * port previosuly make sure to turn off vdd there while
-- 
1.7.9.5

